packages/core/src/interpreter.ts

```diff
@@ -275,6 +275,13 @@ export class Interpreter<
     const isDone = isInFinalState(state.configuration || [], this.machine);
 
     if (this.state.configuration && isDone) {
+      // exit interpreter procedure: https://www.w3.org/TR/scxml/#exitInterpreter
+      this.state.configuration.forEach((stateNode) => {
```

Comment:
Those exit handlers should be executed in "exit order". I'm not sure if `this.state.configuration` holds any guarantee about the order of items in it. It would actually be quite cool to rely on it without having to sort stuff here, I'm just not sure if we can safely make such an assumption now.

Reply:
In the `next` branch, these are sorted by exit order (I think there are some SCXML tests ensuring this). I'll see what I can do here to have the same behavior.

Reply:
The only SCXML I could find regarding this is: https://www.w3.org/Voice/2013/scxml-irp/250/test250.txml and it's classified as manual (it's one of the tests that requires the tester to look at the log output)

Reply:
This would be better suited for the `next` branch, since that more faithfully follows the SCXML algorithm. There would probably need to be substantial changes in order to ensure exit action order; let's scope this PR to just ensuring that they _are_ called.